src/com/phonebookservice/server/FileDataAccessAdapter.java

```diff
@@ -1,13 +1,81 @@
 package com.phonebookservice.server;
 
+import com.phonebookservice.config.Config;
+import com.phonebookservice.config.Config.ConfigKey;
+import com.phonebookservice.database.ContactsDatabase;
+import com.phonebookservice.exception.BadRequestException;
 import com.phonebookservice.model.Contact;
+import com.phonebookservice.util.ErrorCode;
+import com.phonebookservice.util.ErrorMessages;
+import com.phonebookservice.util.StringUtility;
 
 /**
  * file data access adapter.
  *
  * @author Marwa Saleh
  */
-public class FileDataAccessAdapter implements IDataAccessAdapter<Contact> {
+public final class FileDataAccessAdapter
+        implements IDataAccessAdapter<Contact> {
+
+    private static FileDataAccessAdapter singleton;
+    private final ContactsDatabase contactsDb;
+
+    /**
+     * Initialization of file data access adapter.
+     *
+     * @param databaseLink the database link.
+     */
+    private FileDataAccessAdapter(final String databaseLink) {
+        if (StringUtility.isNullOrEmptyString(databaseLink)
+                || !databaseLink.equals(ConfigKey.FILENAME_PATH.getKey())) {
+            throw new BadRequestException(
```

Comment:
Should be an InternalServerError because it is a bad from developer not a client because the developer who is editing in configuration file